src/formatDate.js

```diff
@@ -50,7 +50,35 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const parsedDate = {};
+  let finalDate = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const dataType = fromFormat[i][0].toLowerCase();
+
+    parsedDate[dataType] = date.split(fromFormat[3])[i];
+
+    if (dataType === 'y') {
+      parsedDate[dataType] = parsedDate[dataType].slice(-2);
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const dataType = toFormat[i][0].toLowerCase();
+
+    if (dataType === 'y' && toFormat[i].length === 4) {
+      finalDate += +parsedDate[dataType] < 30
```

Comment:
The code uses the magic number '30'. Instead of hardcoding this value, consider creating a named constant to enhance code readability and maintainability. And in general, try to avoid using mystery numbers, please